src/main/java/transact/logic/commands/DeleteTransactionCommand.java

```diff
@@ -0,0 +1,69 @@
+package transact.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import transact.commons.core.index.Index;
+import transact.commons.util.ToStringBuilder;
+import transact.logic.Messages;
+import transact.logic.commands.exceptions.CommandException;
+import transact.model.Model;
+import transact.model.transaction.Transaction;
+
+/**
+ * Deletes a person identified using it's displayed index from the address book.
+ */
+public class DeleteTransactionCommand extends Command {
+
+    public static final String COMMAND_WORD = "del";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes transaction identified by the index number used in the transaction list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_DELETE_TRANSACTION_SUCCESS = "Deleted Transaction: %1$s";
+
+    private final Index targetIndex;
+
+    public DeleteTransactionCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Transaction> lastShownList = model.getFilteredTransactionList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TRANSACTION_DISPLAYED_INDEX);
+        }
+
+        Transaction transactionToDelete = lastShownList.get(targetIndex.getZeroBased());
+        model.deleteTransaction(transactionToDelete);
+        return new CommandResult(String.format(MESSAGE_DELETE_TRANSACTION_SUCCESS, Messages.format(transactionToDelete)), 2);
```

Comment:
The number is the tab to view (0: Overview, 1: Transactions, 2: person list)
```suggestion
        return new CommandResult(String.format(MESSAGE_DELETE_TRANSACTION_SUCCESS, Messages.format(transactionToDelete)), 1);
```

Reply:
Ill rename the field to an enum later on